ddl/index.go

```diff
@@ -1127,7 +1132,12 @@ func (w *addIndexWorker) batchCheckUniqueKey(txn kv.Transaction, idxRecords []*i
 		} else if w.distinctCheckFlags[i] {
 			// The keys in w.batchCheckKeys also maybe duplicate,
 			// so we need to backfill the not found key into `batchVals` map.
-			val, err := w.index.GenIndexValue(stmtCtx, idxRecords[i].vals, w.distinctCheckFlags[i], false, idxRecords[i].handle)
+			var rsData []types.Datum
+			if t, ok := w.table.(*tables.TableCommon); ok {
+				rsData = t.TryGetHandleRestoredDataWrapper(idxRecords[i].vals, nil)
+			}
```

Comment:
How about the partitioned table?

```sql
drop table if exists t;
create table t (a varchar(255) COLLATE utf8_general_ci primary key clustered, b int) partition by range columns(a) 
(partition p0 values less than ('0'), partition p1 values less than MAXVALUE);
alter table t add index b;
insert into t values ('0', 1);
```

```sql
select * from t use index(b);

+---+------+
| a | b    |
+---+------+
| NULL |    1 |
+---+------+
```

```sql
select * from t ignore index(b);

+---+------+
| a | b    |
+---+------+
| 0 |    1 |
+---+------+
```

Reply:
I think `TryGetHandleRestoredDataWrapper` can be refactored to accept a `table.Table` interface as an argument.